lib/customer-deposit-wallet/http/Cardano/Wallet/Deposit/HTTP/Types/JSON.hs

```diff
@@ -0,0 +1,104 @@
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE StandaloneDeriving #-}
+
+-- |
+-- Copyright: © 2024 Cardano Foundation
+-- License: Apache-2.0
+--
+-- Data types with a JSON schema.
+--
+module Cardano.Wallet.Deposit.HTTP.Types.JSON
+    ( ApiT (..)
+
+    -- * Re-exports
+    , Address
+    , Customer
+    , CustomerList
+    )
+    where
+
+import Prelude
+
+import Cardano.Wallet.Deposit.HTTP.Types.JSON.Encoding
+    ( ViaText (..)
+    , customOptions
+    )
+import Cardano.Wallet.Deposit.Pure
+    ( Customer
+    )
+import Cardano.Wallet.Deposit.Read
+    ( Address
+    )
+import Data.Aeson
+    ( FromJSON (..)
+    , ToJSON (..)
+    , genericParseJSON
+    , genericToJSON
+    )
+import Data.Bifunctor
+    ( bimap
+    , first
+    )
+import Data.Text
+    ( Text
+    )
+import Data.Text.Class
+    ( FromText (..)
+    , getTextDecodingError
+    )
+import Servant
+    ( FromHttpApiData (..)
+    )
+
+import qualified Data.Text as T
+
+{-----------------------------------------------------------------------------
+    Additional type definitions
+------------------------------------------------------------------------------}
+
+type CustomerList = [(Customer, Address)]
+
+{-----------------------------------------------------------------------------
+   ApiT
+------------------------------------------------------------------------------}
+
+newtype ApiT a = ApiT {unApiT :: a}
```

Comment:
this might be located in the end somewhere else as it could not only refer to JSON ... maybe in `Cardano.Wallet.Deposit.HTTP.Types` module. `CustomerList` would go there too.

Reply:
Even though the name is general, I would like to keep `ApiT` specific to JSON and URL paths, so this module is a good place for now.

(I also want to avoid the situation where a child module `A.B.C` to import a parent module `A.B`.)

The `CustomerList` type is mainly a helpful abbreviation for use in the `*.HTTP.Types.API` module. If we really care about this definition, I would put it into `Cardano.Wallet.Deposit.IO` instead.

Reply:
Oh, and I have to define the instances of `ApiT …` in the same module as the type `ApiT`, otherwise they are orphaned instances. 🤓